topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -61,23 +74,51 @@
     /*test*/ ZkTopicWatcher topicWatcher;
     /*test*/ PrometheusMeterRegistry metricsRegistry;
     K8sTopicWatcher watcher;
-    /** The id of the periodic reconciliation timer. This is null during a periodic reconciliation. */
+    /**
+     * The id of the periodic reconciliation timer. This is null during a periodic reconciliation.
+     */
     private volatile Long timerId;
     private volatile boolean stopped = false;
     private Zk zk;
     private volatile HttpServer healthServer;
 
+    /**
+     * @param kubeClient kubernetes client
+     * @param config     TopicOperator config
+     */
     public Session(KubernetesClient kubeClient, Config config) {
+        this(kubeClient, config, null, null, new TopicOperatorState());
+    }
+
+    /**
+     * Constructor is package private for testing
+     *
+     * @param kubeClient         kubernetes client
+     * @param config             TopicOperator config
+     * @param topicStoreCreator  indirection to allow unit testing of start-up sequence with mock topicstore creation
+     * @param zkClientCreator    indirection to allow unit testing of start-up sequence with mock ZK client creation
+     * @param topicOperatorState used to communicate liveness/readiness to health server from things that can potentially fail
+     */
+    Session(KubernetesClient kubeClient, Config config, BiFunction<Zk, Config, TopicStore> topicStoreCreator, BiFunction<Vertx, Config, Future<Zk>> zkClientCreator, TopicOperatorState topicOperatorState) {
+        this.topicStoreCreator = topicStoreCreator == null ? this::createTopicStore : topicStoreCreator;
+        this.zkClientCreator = zkClientCreator == null ? this::createZk : zkClientCreator;
         this.kubeClient = kubeClient;
         this.config = config;
+        this.topicOperatorState = topicOperatorState;
         StringBuilder sb = new StringBuilder(System.lineSeparator());
-        for (Config.Value<?> v: Config.keys()) {
+        for (Config.Value<?> v : Config.keys()) {
             sb.append("\t").append(v.key).append(": ").append(Util.maskPassword(v.key, config.get(v).toString())).append(System.lineSeparator());
         }
         LOGGER.info("Using config:{}", sb.toString());
         this.metricsRegistry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
     }
 
+    @Override
+    public void init(Vertx vertx, Context context) {
+        super.init(vertx, context);
+        executor = vertx.createSharedWorkerExecutor("blocking-startup-ops", 1);
```

Comment:
@fvaleri took your advice, and moved worker pool creation inside :)

Reply:
Great, much cleaner and safer.